### PR TITLE
[RDY] Fix unit tests for NPC needs

### LINKED ISSUE #19022: on_load-sane-values unit test failures
Using 9d41e9fd4a4c453b799faa43a60d469295cf8eff

```
-------------------------------------------------------------------------------
on_load-sane-values
  Awake for 10 minutes, gaining hunger/thirst/fatigue
-------------------------------------------------------------------------------
npc_test.cpp:41
...............................................................................

npc_test.cpp:55: FAILED:
  CHECK( test_npc.get_hunger() >= five_min_ticks - margin )
with expansion:
  0 >= 1

npc_test.cpp:56: FAILED:
  CHECK( test_npc.get_thirst() >= five_min_ticks - margin )
with expansion:
  0 >= 1

-------------------------------------------------------------------------------
on_load-sane-values
  Awake for 2 days, gaining hunger/thirst/fatigue
-------------------------------------------------------------------------------
npc_test.cpp:41
...............................................................................

npc_test.cpp:69: FAILED:
  CHECK( test_npc.get_hunger() >= five_min_ticks - margin )
with expansion:
  147 >= 566 (0x236)

npc_test.cpp:70: FAILED:
  CHECK( test_npc.get_thirst() >= five_min_ticks - margin )
with expansion:
  151 >= 566 (0x236)

-------------------------------------------------------------------------------
on_load-sane-values
  Sleeping for 6 hours, gaining hunger/thirst (not testing fatigue due to lack
  of effects processing)
-------------------------------------------------------------------------------
npc_test.cpp:41
...............................................................................

npc_test.cpp:93: FAILED:
  CHECK( test_npc.get_hunger() >= expected_change - margin )
with expansion:
  9 >= 26

npc_test.cpp:94: FAILED:
  CHECK( test_npc.get_thirst() >= expected_change - margin )
with expansion:
  9 >= 26
```

### PR DESCRIPTION
**Pull request by codemime**
This PR is based on #18461 (collaboration with @mugling) and aims (probably reaches) two goals for now:
- ~~Support displaying standard `catch` help without creating a test world~~ Factored out
- Fix unit tests for NPC needs (hunger and thirst were brought back, quartered and capped in #15759). No tests for the caps yet.

I also fiddled with `margin`s a bit: needs are updated using `roll_remainder()` and hence hunger and thirst have some variety. More adjustments may be required.

EDIT: Fixes #19022.

### CODE REVIEW

**Comment by mugling** [comment]
Can you factor out d2674cd as a separate PR as that is a simple but useful change

File: tests/npc_test.cpp [hunk 1]
```diff
@@ -7,8 +7,23 @@
...
+{
+    range() :
+        min( std::numeric_limits<Val>::min() ),
+        max( std::numeric_limits<Val>::max() ) {}
+    range( Val value, Val margin ) :
+        min( value - margin ),
+        max( value + margin ) {}
+
+    Val min;
+    Val max;
```
Comment (mugling) [inline tests/npc_test.cpp]: I'd probably define an operator that compares a value to the ranges so you can cut the number of `CHECK` statements by half.

File: tests/npc_test.cpp [hunk 2]
```diff
@@ -7,8 +7,23 @@
...
 #include <string>
 
+template<typename Val>
+struct range
+{
+    range() :
+        min( std::numeric_limits<Val>::min() ),
+        max( std::numeric_limits<Val>::max() ) {}
+    range( Val value, Val margin ) :
+        min( value - margin ),
```
Comment (codemime) [inline tests/npc_test.cpp]: No unsigned numbers yet, but in the future - yes. Fixed.

**Comment by codemime** [comment]
All set.

**Comment by kevingranade** [comment]
Looking good, will build and test.

**Pull request merged**